pkg/kube/networkpolicy.go

```diff
@@ -0,0 +1,206 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kube
+
+import (
+	"context"
+
+	"github.com/istio-ecosystem/sail-operator/pkg/config"
+	"github.com/istio-ecosystem/sail-operator/pkg/constants"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var log = ctrl.Log.WithName("netpol")
+
+// CreateNetworkPolicy creates a NetworkPolicy for the sail-operator during startup
+// Uses a direct clientset since this runs before the manager cache is started
+func CreateNetworkPolicy(cfg *rest.Config, namespace string, platform config.Platform) error {
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	networkPolicy := buildNetworkPolicy(namespace, platform)
+
+	ctx := context.Background()
```

Comment:
Does this need a timeout? I think these get/create requests will block the operator from reconciling resources until these requests complete.

Actually I think you'll want to inherit from the context returned by: `ctrl.SetupSignalHandler` so that these requests will be canceled when interrupt is given. So can you pass that context in here and add a reasonable timeout to it?